plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/AggregationQueryPageSource.java

```diff
@@ -0,0 +1,187 @@
+import static io.trino.plugin.elasticsearch.ElasticsearchQueryBuilder.buildAggregationQuery;
+import static io.trino.plugin.elasticsearch.ElasticsearchQueryBuilder.buildSearchQuery;
+import static java.util.Objects.requireNonNull;
+
+public class AggregationQueryPageSource
+        implements ConnectorPageSource
+{
+    private final long readTimeNanos;
+    private final SearchResponse searchResponse;
+    private final List<Decoder> decoders;
+    private final BlockBuilder[] columnBuilders;
+    private final List<ElasticsearchColumnHandle> columns;
+    private boolean fetched;
+
+    public AggregationQueryPageSource(
+            ElasticsearchClient client,
+            ElasticsearchTableHandle table,
+            ElasticsearchSplit split,
+            List<ElasticsearchColumnHandle> columns)
+    {
+        requireNonNull(client, "client is null");
+        requireNonNull(table, "table is null");
+        requireNonNull(split, "split is null");
+        requireNonNull(columns, "columns is null");
+        this.columns = columns;
+        columnBuilders = columns.stream()
+                .map(ElasticsearchColumnHandle::getType)
+                .map(type -> type.createBlockBuilder(null, 1))
+                .toArray(BlockBuilder[]::new);
+        decoders = createDecoders(columns);
+
+        long start = System.nanoTime();
+        this.searchResponse = client.beginSearch(
+                split.getIndex(),
+                split.getShard(),
+                buildSearchQuery(table.getConstraint().transform(ElasticsearchColumnHandle.class::cast), table.getQuery()),
+                Optional.ofNullable(buildAggregationQuery(table.getAggTerms(), table.getAggregates())),
+                Optional.empty(),
+                Collections.emptyList(),
+                Optional.empty(),
+                table.getLimit());
+        readTimeNanos = System.nanoTime() - start;
+    }
+
+    @Override
+    public long getCompletedBytes()
+    {
+        return 0;
+    }
+
+    @Override
+    public long getReadTimeNanos()
+    {
+        return readTimeNanos;
+    }
+
+    @Override
+    public boolean isFinished()
+    {
+        return fetched;
+    }
+
+    @Override
+    public Page getNextPage()
+    {
+        fetched = true;
+        final SearchHit noUsed = new SearchHit(0);
+        List<Map<String, Object>> flatResult = flattern(searchResponse.getAggregations());
+        for (Map<String, Object> result : flatResult) {
+            for (int i = 0; i < columns.size(); i++) {
+                String key = columns.get(i).getName();
+                decoders.get(i).decode(noUsed, () -> result.get(key), columnBuilders[i]);
+            }
+        }
+        Block[] blocks = new Block[columnBuilders.length];
+        for (int i = 0; i < columnBuilders.length; i++) {
+            blocks[i] = columnBuilders[i].build();
+            columnBuilders[i] = columnBuilders[i].newBlockBuilderLike(null);
```

Comment:
Why is this needed?

Reply:
just copy from `io.trino.plugin.elasticsearch.ScanQueryPageSource`